pkgs/development/python-modules/pre-commit/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib, buildPythonPackage, fetchPypi, pythonOlder
```

Comment:
This is already packaged as an application https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/version-management/git-and-tools/pre-commit/default.nix
Though this has some nice improvements.

Since you've went ahead and packaged this as a library and it already exists as an application, perhaps we should use `toPythonApplication` for `pre-commit`? See the nixpkgs manual for explanation.

If not I would go ahead and merge these changes somehow into the current `pre-commit` derivation.